design/authenticators_api.md

```diff
@@ -0,0 +1,178 @@
+
+3. enabled: these authenticators are whitelisted in the CONJUR_AUTHENTICATORS variable.
+
+Section 1 has important data but it shouldn't be here, but rather in the docs. It has
+nothing to do with the given Conjur instance and just describes the authenticators that
+are implemented in Conjur.
+
+Sections 2 & 3 combined can help a conjur user to know if he can use an authenticator
+but that's not entirely true. some authenticators need further configuration than 
+just configuring a webservice and enabling the authenticator in the ENV (for example, 
+authn-oidc needs to load variables). So while a user will get _some_ value using this API, 
+it won't fully let him know that he can use the authenticator, which is the purpose of this.
+
+## Feature Overview 
+
+### Use Cases
+
+This feature is divided into 2 personas:
+
+#### Conjur operator
+
+**As a** Conjur operator\
+**I'd like to** know which authenticators are configured correctly\
+**So that** I can complete the configuration properly if it's incomplete (or remove
+it if it's not needed)
+
+#### Conjur user
+
+**As a** Conjur user\
+**I'd like to** know which authenticators are available for authentication\
+**So that** I can authenticate with a properly configured one
+
+Note that both personas would like to know the same thing (configured correctly == available for authentication)
+but the action they need it for is different. 
+
+We can learn 2 things from the above:
+
+1. The Conjur user needs to authenticate according to the response. So this request must be applicable **without** a Conjur access token. 
+1. The Conjur operator needs to know which actions are needed for completing the configuration,
+which may consist sensitive data. So this request must be applicable **only with** a Conjur access token.
+
+This leads us to 2 API endpoints:
+
+1. `/authenticators` (for Conjur users)
+2. `/authenticators/health` (for Conjur operators)
+
+### Process Logic
+
+In the following example, the authenticators `authn` & `authn-1` are properly configured
+and `authn-2` is not whitelisted in the ENV.
+
+#### `/authenticators` Endpoint
+
+- A Conjur user runs an `/authenticators` request:\
+`GET /authenticators`
+- The user gets a response with code 200 with the following body:
+```
+{
+   “authenticators”:
+   [
+      "authn",
+      "authn-1"
+   ]
+}
+```
+
+#### `/authenticators/health` Endpoint
+
+- A Conjur operator logs into Conjur (in any authn method) and receives an access token
+- The operator runs an `/authenticators/health` request with the given access token:\
+`GET /authenticators/health`
+- The operator gets a response with code 200 (more info [here](authenticators_health_api.md#response-code-for-unhealthy-authenticators)) with the following body:
+```
+{
+   “authenticators”:
+   {
+      "authn": "ok",
+      "authn-1": "ok",
+      "authn-2": "Authentication::Security::NotWhitelisted: CONJ00004E 'authn-2' is not whitelisted in CONJUR_AUTHENTICATORS", 
+   },
+   "ok": false
+}
```

Comment:
If we do end up having an aggregated list, the custom format might not be advisable as you then need harder logic to monitor the endpoint. I think we probably should lean toward something easier for machine processing:
```json
{
   "authenticators":
   {
      "authn": {
        "status": "ok"
      },
      "authn-1": {
        "status": "ok"
      },
      "authn-2": {
        "status": "error",
        "error": "Authentication::Security::NotWhitelisted",
        "code": "CONJ00004E",
        "message": "'authn-2' is not whitelisted in CONJUR_AUTHENTICATORS",
      }
   },
   "status": "error"
}
```

Reply:
i like it. changed in the doc